core/src/main/scala/org/apache/spark/status/ElementTrackingStore.scala

```diff
@@ -46,7 +50,26 @@ import org.apache.spark.util.kvstore._
  */
 private[spark] class ElementTrackingStore(store: KVStore, conf: SparkConf) extends KVStore {
 
-  private val triggers = new HashMap[Class[_], Seq[Trigger[_]]]()
+  private class LatchedTriggers(val triggers: Seq[Trigger[_]]) {
+    val countDeferred = new AtomicInteger(0)
```

Comment:
You're kinda using this as a boolean, so `AtomicBoolean`? Also, `private`, and the variable name is a little cryptic (maybe just call it `busy` or something?).